File: src/main/java/cn/booklish/sharp/client/util/RpcRequestIdGenerator.java
package cn.booklish.sharp.client.util;

import java.util.concurrent.atomic.AtomicInteger;

/**
 * @Author: liuxindong
 * @Description: Rpc请求的表示id生成工具
 * @Create: 2017/11/23 15:53
 * @Modify:
 */
public class RpcRequestIdGenerator {

    private static final AtomicInteger integer = new AtomicInteger();

    public static Integer getId(){
        return integer.getAndIncrement();
    }

}

File: src/main/java/cn/booklish/sharp/proxy/RpcRequestIdGenerator.kt
package cn.booklish.sharp.proxy

import java.util.concurrent.atomic.AtomicInteger

/**
 * Rpc请求id生成器
 */
object RpcRequestIdGenerator {

    private val currentId = AtomicInteger()

    fun getId(): Int {
        currentId.compareAndSet(Int.MAX_VALUE, Int.MIN_VALUE)
        return currentId.getAndIncrement()
    }

}